src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,322 +78,538 @@ private static VersionInfo GetVersionInfo()
         internal const string FriendlyName = "NA Indicator Transform";
         internal const string ShortName = "NAInd";
 
-        private static string TestType(ColumnType type)
-        {
-            // Item type must have an NA value. We'll get the predicate again later when we're ready to use it.
-            Delegate del;
-            if (Conversions.Instance.TryGetIsNAPredicate(type.ItemType, out del))
-                return null;
-            return string.Format("Type '{0}' is not supported by {1} since it doesn't have an NA value",
-                type, LoaderSignature);
-        }
-
-        private const string RegistrationName = "NaIndicator";
+        private const string RegistrationName = nameof(NAIndicatorTransform);
 
-        // The output column types, parallel to Infos.
-        private readonly ColumnType[] _types;
+        internal (string input, string output)[] GetColumnPairs() => ColumnPairs;
```

Comment:
>internal [](start = 8, length = 8)

The pattern is :
`public IReadOnlyList<(string input, string output)> Columns => ColumnPairs.AsReadOnly()`